### ISSUE TITLE
Fix RKE-CI tests due to configInline deprecation from MetalLB

### ISSUE THREAD

**Issue by mmartin24** [epinio/epinio#1626]
Note: this part has been edited on 07/12/22

The initial scope of this ticket was to investigate why RKE-CI tests started to fail some days ago. We see from 07/08/22 a this consistent error:

```
 Release "metallb" does not exist. Installing it now.
  Error: execution error at (metallb/templates/deprecated_configInline.yaml:2:4): Starting from v0.13.0 configInline is no longer supported. Please see https://metallb.universe.tf/#backward-compatibility
```

The problem is this [PR](https://github.com/metallb/metallb/pull/1485) from MetalLB where they deprecated the `configInline` file . Having identified this, we need to reconvert that file into something else in order to make it work.

**Comment by mmartin24** [comment]
Today error seems to be different: https://github.com/epinio/epinio/runs/7246759009?check_suite_focus=true#step:12:144

```
Release "metallb" does not exist. Installing it now.
  Error: execution error at (metallb/templates/deprecated_configInline.yaml:2:4): Starting from v0.13.0 configInline is no longer supported. Please see https://metallb.universe.tf/#backward-compatibility
  Unexpected error:
      <*exec.ExitError | 0xc0007bf520>: {
          ProcessState: {
              pid: 2280,
              status: 256,
              rusage: {
                  Utime: {Sec: 0, Usec: 456488},
                  Stime: {Sec: 0, Usec: 188919},
                  Maxrss: 51812,
                  Ixrss: 0,
                  Idrss: 0,
                  Isrss: 0,
                  Minflt: 27[98](https://github.com/epinio/epinio/runs/7246759009?check_suite_focus=true#step:12:99),
                  Majflt: 0,
                  Nswap: 0,
                  Inblock: 0,
                  Oublock: [101](https://github.com/epinio/epinio/runs/7246759009?check_suite_focus=true#step:12:102)6,
                  Msgsnd: 0,
                  Msgrcv: 0,
                  Nsignals: 0,
                  Nvcsw: 2765,
                  Nivcsw: 209,
              },
          },
          Stderr: nil,
      }
      exit status 1
  occurred
In [It] at: /home/sles/actions-runner/_work/epinio/epinio/acceptance/install/scenario3_test.go:84
```
I am not sure if it is needed an update on the test in regards to `configInline`

**Comment by mmartin24** [comment]
I investigated this a bit more.
In the release notes 0.13.3 from https://metallb.universe.tf/release-notes/ they force the failure of helm release if the a configInline file is provided: https://github.com/metallb/metallb/pull/1485. Since we have a file like that failure is expected and we need to substitute it.

In our case the file seems to be located here: `/epinio/assets/tests/values-metallb-rke.yml`

```
configInline:
  address-pools:
  - name: default
    protocol: layer2
    addresses:
    - @IP_RANGE@
```

Now, about what to do, I am not sure yet but this is what I have come so far in terms of possible paths of solutions:

**Path 1**:
The error redirect to this page (https://metallb.universe.tf/#backward-compatibility) but I am not sure I find anything useful there. If I understand correctly it is showing a way of backcompatiblity via configmap. If what it implies is that we use and old way, I found this [link](https://v0-7-3--metallb.netlify.app/tutorial/layer2/) on v.0.7.3 on how to configure a an sample MetaLB configuration in example-layer2-config.yaml:

```
apiVersion: v1
kind: ConfigMap
metadata:
  namespace: metallb-system
  name: config
data:
  config: |
    address-pools:
    - name: my-ip-space
      protocol: layer2
      addresses:
      - 192.168.1.240-192.168.1.250
```

**Path 2**
On the most recent version MetalLB v.0.13.3 they show the following advanced L2 configuration. I am not sure if the path is adapting somehow our data to this: https://metallb.universe.tf/configuration/_advanced_l2_configuration/)
```
apiVersion: metallb.io/v1beta1
kind: L2Advertisement
metadata:
  name: example
spec:
  ipAddressPools:
  - second-pool
  nodeSelectors:
  - matchLabels:
      kubernetes.io/hostname: NodeC
```
@andreas-kupries , could you please help me out / give a bit of guidance here?
Thanks

**Comment by mmartin24** [comment]
I turned out to be wrong. 
After a nice conversation with @enrichman and @andreas-kupries what seems to be needed is explained in https://metallb.universe.tf/#backward-compatibility:

> In order to use the tool the container must be ran mapping the path with the source file config.yaml to /var/input, and it will generate a resources.yaml file containing the resources mapped to the yaml.
`docker run -d -v $(pwd):/var/input quay.io/metallb/configmaptocrs -source config.yaml `

I applied this locally using:
```
➜  tests git:(main) ✗ docker run -d -v $(pwd):/var/input quay.io/metallb/configmaptocrs -source values-metallb-rke.yml
97a5835e9366ed95b9d8ff8ad1516621afb4976129e9a73fdf9bc9ef2bd3030d
```
However, the `resources.yaml` file created is empty:
![Image](https://user-images.githubusercontent.com/37271841/178473533-997e0efc-477e-4f40-9926-6f532a4ecdeb.png)

Does anybody know if this correct? If not I will try it in a PR
Thanks

**Comment by andreas-kupries** [comment]
Looking at `assets/tests/values-metallb-rke.yml` again I see that while it looks like yaml, it does not look like a full kubernetes resource. More like a fragment. ... Looking further ...

> out, err = proc.RunW("helm", "upgrade", "--install", "--wait", "-n", "metallb",
                                "--create-namespace", "metallb", "metallb/metallb", "-f",
                                testenv.TestAssetPath("values-metallb-rke.yml"))

is the using place (one of them) ... This file serves as a `values.yaml` for the metallb helm chart.
Our thought wrt `docker run` to convert are completely wrong.
Apologies.

Reread https://metallb.universe.tf/#backward-compatibility ... I am also unable to find any reference to the `configInLine` on that page. I will ask on the MetalLB PR disabling configinline about that.

**Comment by mmartin24** [comment]
Thank you very much @andreas-kupries for keep helping here and for all the input, even if we are wrong.
Let's wait then a bit until further response.

**Comment by mmartin24** [comment]
thank you @andreas-kupries ,
I will try then what you comment above and will redirect the test towards **resources.yaml** just created. Definitively trying with a real ip helped to create the file.
![image](https://user-images.githubusercontent.com/37271841/178509247-01134f14-34ce-4380-b7c6-e8754f7f52b2.png)


I will comment here once I have the results

**Comment by mmartin24** [comment]
I updated the ticket title and initial content after the scope is clear.

**Comment by mmartin24** [comment]
Quick status: after a bit of tweaking, it seems the issue with the configInline may be fixed as commented in the [PR](https://github.com/epinio/epinio/pull/1637#issuecomment-1182875942); however seems to fail on later steps. I will continue taking a look.

**Comment by mmartin24** [comment]
Finally, it seems fixed in the pr: #1637
